master/internal/api_runs.go

```diff
@@ -403,3 +405,126 @@ func (a *apiServer) MoveRuns(
 	}
 	return &apiv1.MoveRunsResponse{Results: results}, nil
 }
+
+func (a *apiServer) ArchiveRuns(
+	ctx context.Context, req *apiv1.ArchiveRunsRequest,
+) (*apiv1.ArchiveRunsResponse, error) {
+	results, err := archiveUnarchiveAction(ctx, true, req.RunIds, req.ProjectId, req.Filter)
+	if err != nil {
+		return nil, err
+	}
+	return &apiv1.ArchiveRunsResponse{Results: results}, nil
+}
+
+func (a *apiServer) UnarchiveRuns(
+	ctx context.Context, req *apiv1.UnarchiveRunsRequest,
+) (*apiv1.UnarchiveRunsResponse, error) {
+	results, err := archiveUnarchiveAction(ctx, false, req.RunIds, req.ProjectId, req.Filter)
+	if err != nil {
+		return nil, err
+	}
+	return &apiv1.UnarchiveRunsResponse{Results: results}, nil
+}
+
+func archiveUnarchiveAction(ctx context.Context, archive bool, runIDs []int32,
+	projectID int32, filter *string,
+) ([]*apiv1.RunActionResult, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var runChecks []archiveRunOKResult
+	query := db.Bun().NewSelect().
+		ModelTableExpr("runs AS r").
+		Model(&runChecks).
+		Column("r.id").
+		Column("r.archived").
+		ColumnExpr("r.experiment_id as exp_id").
+		ColumnExpr("false as is_multitrial").
+		ColumnExpr("r.state IN (?) AS state", bun.In(model.StatesToStrings(model.TerminalStates))).
+		Join("LEFT JOIN experiments e ON r.experiment_id=e.id").
+		Join("JOIN projects p ON r.project_id = p.id").
+		Join("JOIN workspaces w ON p.workspace_id = w.id").
+		Where("r.project_id = ?", projectID)
+
+	if filter == nil {
+		query = query.Where("r.id IN (?)", bun.In(runIDs))
+	} else {
+		query, err = filterRunQuery(query, filter)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
or, as i'm reading this, since it has `if filter == nil` in a few places, maybe it just makes sense as two functions. and then you try to not repeat yourself some other way.

Reply:
Added error for cases when RunIDs and Filter are both filled. Will add this behaviour for other endpoints in their respective PRs